File: public/js/i18n.js
/**
 * Lightweight i18n. English strings are the keys; dictionaries map them to
 * Arabic (RTL) and French. t(s) falls back to the English string, so pages
 * that are not yet translated keep working. Language persists in localStorage
 * and switching re-renders the app (App.route()).
 *
 * To add a language: add a dictionary below and an option in the topbar select.
 */
const I18N = {
  ar: {
    // sections
    'General': 'عام', 'Material Requests': 'طلبات المواد', 'Warehouse Execution': 'تنفيذ المستودع',
    'Receiving & Quality': 'الاستلام والجودة', 'Inventory': 'المخزون', 'Master Data': 'البيانات الرئيسية',
    'Administration': 'الإدارة',
    // menu
    'Dashboard': 'لوحة التحكم', 'KPI Dashboard': 'مؤشرات الأداء', 'Notifications': 'الإشعارات',
    'AI Stock Analytics': 'تحليلات المخزون الذكية',
    'Create Request': 'إنشاء طلب', 'Requests': 'الطلبات', 'Approvals': 'الموافقات', 'ERP Operator': 'مشغل ERP',
    'Warehouse Dashboard': 'لوحة المستودع', 'Bin & Batch Assign': 'تخصيص الموقع والدفعة',
    'Picker Assignment': 'تعيين المنفذ', 'My Picking Tasks': 'مهام الالتقاط', 'Goods Issue Posting': 'ترحيل الصرف',
    'Goods Receipt & QR': 'استلام البضائع و QR', 'QR Label Printing': 'طباعة ملصقات QR',
    'Batch Tracking': 'تتبع الدفعات', 'Expiry Alerts': 'تنبيهات الصلاحية', 'Quality': 'الجودة',
    'All Locations': 'كل المواقع', 'Empty Locations': 'المواقع الفارغة',
    'Materials': 'المواد', 'Locations': 'المواقع', 'Warehouses': 'المستودعات', 'Bin Locations': 'مواقع التخزين',
    'Movement Types': 'أنواع الحركة', 'Audit Trail': 'سجل التدقيق', 'Users Management': 'إدارة المستخدمين',
    'Permissions': 'الصلاحيات',
    // titles
    'Create Material Request': 'إنشاء طلب مواد', 'Request Detail': 'تفاصيل الطلب',
    'Manager Approvals': 'موافقات المدير', 'ERP Operator Queue': 'قائمة مشغل ERP',
    'Bin & Batch Assignment': 'تخصيص الموقع والدفعة', 'Quality Management': 'إدارة الجودة',
    'Warehouse Master': 'المستودعات الرئيسية', 'Bin Location Master': 'مواقع التخزين الرئيسية',
    'Movement Type Config': 'إعداد أنواع الحركة', 'Permissions Management': 'إدارة الصلاحيات',
    // common
    'Logout': 'تسجيل الخروج', 'Login': 'تسجيل الدخول', 'Sign up': 'إنشاء حساب', 'Email': 'البريد الإلكتروني',
    'Password': 'كلمة المرور', 'Full name': 'الاسم الكامل', 'Save': 'حفظ', 'Cancel': 'إلغاء',
    'Language': 'اللغة', 'Theme': 'المظهر', 'Warehouse Management System': 'نظام إدارة المستودعات',
    'Loading…': 'جارٍ التحميل…',
    // common actions
    'Submit': 'إرسال', 'Delete': 'حذف', 'Edit': 'تعديل', 'Add': 'إضافة', 'Close': 'إغلاق', 'Back': 'رجوع',
    'Search': 'بحث', 'Reset': 'إعادة تعيين', 'Export': 'تصدير', 'Download': 'تنزيل', 'Upload': 'رفع',
    'Reason': 'السبب', 'Confirm': 'تأكيد', 'Approve': 'موافقة', 'Reject': 'رفض', 'Return': 'إرجاع',
    'Status': 'الحالة', 'Quantity': 'الكمية', 'Actions': 'إجراءات', 'Details': 'التفاصيل',
    // new features (P2/P3)
    'Cycle Counting': 'الجرد الدوري', 'Attachments': 'المرفقات', 'Reverse GI': 'عكس الصرف',
    'Reverse Goods Issue': 'عكس صرف البضائع', 'High-Value Approvals': 'موافقات عالية القيمة',
    'Counted quantity': 'الكمية المعدودة', 'Variance': 'الفرق', 'System quantity': 'كمية النظام',
    'Upload file': 'رفع ملف', 'No attachments': 'لا توجد مرفقات',
      // navigation-v2 group headings and route labels — the frame around every
    // screen. These are the first thing a storekeeper reads and the last thing
    // that should be in English.
    'Command Center': 'مركز التحكم', 'Demand & Requests': 'الطلبات',
    'Inbound Operations': 'عمليات الاستلام', 'Outbound Operations': 'عمليات الصرف',
    'Inventory Control': 'مراقبة المخزون', 'Intelligence & Analytics': 'التحليلات',
    'Master Data & Integration': 'البيانات الرئيسية والتكامل',
    'Governance & Administration': 'الحوكمة والإدارة',
    'Subcontractor Materials': 'مواد مقاولي الباطن',
    'Operations Overview': 'نظرة عامة على العمليات', 'Performance Cockpit': 'مؤشرات الأداء',
    'Alerts & Notifications': 'التنبيهات والإشعارات',
    'Request Work Queue': 'قائمة الطلبات',
    'Approval Work Queue': 'قائمة الموافقات', 'ERP Processing Queue': 'قائمة معالجة ERP',
    'Goods Receipt & Identification': 'استلام المواد وترميزها', 'QR & Label Printing': 'طباعة الباركود',
    'Quality Inspection': 'فحص الجودة', 'Batch Traceability': 'تتبع الدفعات',
    'Shelf-life & Expiry Control': 'الصلاحية وتواريخ الانتهاء',
    'Execution Control Board': 'لوحة تنفيذ المستودع', 'Bin & Batch Allocation': 'تخصيص الموقع والدفعة',
    'Work Assignment': 'توزيع المهام', 'Picking Tasks': 'مهام الصرف',
    'Stock Reallocation': 'إعادة تخصيص المخزون',
    'Packing, Dispatch & Shipping': 'التعبئة والشحن',
    'Physical Inventory': 'الجرد الفعلي',
    'Stock by Location': 'المخزون حسب الموقع', 'Available Locations': 'المواقع المتاحة',
    'AI Inventory Intelligence': 'تحليلات المخزون الذكية',
    'Material Master': 'أصناف المواد', 'Storage Location Master': 'مواقع التخزين',
    'Bin Master': 'المواقع داخل المستودع',
    'Movement Type Configuration': 'إعداد أنواع الحركة', 'Data Integration Center': 'مركز استيراد البيانات',
    'Audit & Traceability': 'سجل التدقيق والتتبع', 'User Administration': 'إدارة المستخدمين',
    'Roles & Permissions': 'الأدوار والصلاحيات',
    'Deliveries & Quality Inspection': 'التوريدات وفحص الجودة',
    'Subcontractor On-Hand Stock': 'أرصدة مقاولي الباطن',
    'Subcontractor Reconciliation': 'تسوية مقاولي الباطن',
    'Returns to Owner': 'المرتجعات للمالك',
    'Subcontractor-Owned Stock': 'المواد المملوكة لمقاولي الباطن',
    'Subcontractors & Categories': 'مقاولو الباطن والتصنيفات',
},
  fr: {
    'General': 'Général', 'Material Requests': 'Demandes de matériel', 'Warehouse Execution': 'Exécution entrepôt',
    'Receiving & Quality': 'Réception et qualité', 'Inventory': 'Inventaire', 'Master Data': 'Données de base',
    'Administration': 'Administration',
    'Dashboard': 'Tableau de bord', 'KPI Dashboard': 'Tableau KPI', 'Notifications': 'Notifications',
    'AI Stock Analytics': 'Analyse IA du stock',
    'Create Request': 'Créer une demande', 'Requests': 'Demandes', 'Approvals': 'Approbations', 'ERP Operator': 'Opérateur ERP',
    'Warehouse Dashboard': "Tableau d'entrepôt", 'Bin & Batch Assign': 'Affectation casier/lot',
    'Picker Assignment': 'Affectation préparateur', 'My Picking Tasks': 'Mes préparations', 'Goods Issue Posting': 'Sortie de marchandises',
    'Goods Receipt & QR': 'Réception & QR', 'QR Label Printing': 'Impression étiquettes QR',
    'Batch Tracking': 'Suivi des lots', 'Expiry Alerts': 'Alertes de péremption', 'Quality': 'Qualité',
    'All Locations': 'Tous les emplacements', 'Empty Locations': 'Emplacements vides',
    'Materials': 'Articles', 'Locations': 'Emplacements', 'Warehouses': 'Entrepôts', 'Bin Locations': 'Casiers',
    'Movement Types': 'Types de mouvement', 'Audit Trail': "Piste d'audit", 'Users Management': 'Gestion des utilisateurs',
    'Permissions': 'Autorisations',
    'Create Material Request': 'Créer une demande de matériel', 'Request Detail': 'Détail de la demande',
    'Manager Approvals': 'Approbations manager', 'ERP Operator Queue': "File de l'opérateur ERP",
    'Bin & Batch Assignment': 'Affectation casier/lot', 'Quality Management': 'Gestion de la qualité',
    'Warehouse Master': 'Référentiel entrepôts', 'Bin Location Master': 'Référentiel casiers',
    'Movement Type Config': 'Config. types de mouvement', 'Permissions Management': 'Gestion des autorisations',
    'Logout': 'Déconnexion', 'Login': 'Connexion', 'Sign up': "S'inscrire", 'Email': 'E-mail',
    'Password': 'Mot de passe', 'Full name': 'Nom complet', 'Save': 'Enregistrer', 'Cancel': 'Annuler',
    'Language': 'Langue', 'Theme': 'Thème', 'Warehouse Management System': 'Système de gestion d’entrepôt',
    'Loading…': 'Chargement…',
    // common actions
    'Submit': 'Soumettre', 'Delete': 'Supprimer', 'Edit': 'Modifier', 'Add': 'Ajouter', 'Close': 'Fermer', 'Back': 'Retour',
    'Search': 'Rechercher', 'Reset': 'Réinitialiser', 'Export': 'Exporter', 'Download': 'Télécharger', 'Upload': 'Téléverser',
    'Reason': 'Motif', 'Confirm': 'Confirmer', 'Approve': 'Approuver', 'Reject': 'Rejeter', 'Return': 'Retourner',
    'Status': 'Statut', 'Quantity': 'Quantité', 'Actions': 'Actions', 'Details': 'Détails',
    // new features (P2/P3)
    'Cycle Counting': "Comptage tournant", 'Attachments': 'Pièces jointes', 'Reverse GI': 'Annuler la sortie',
    'Reverse Goods Issue': 'Annuler la sortie de marchandises', 'High-Value Approvals': 'Approbations à forte valeur',
    'Counted quantity': 'Quantité comptée', 'Variance': 'Écart', 'System quantity': 'Quantité système',
    'Upload file': 'Téléverser un fichier', 'No attachments': 'Aucune pièce jointe',
  },
};

const RTL_LANGS = ['ar'];

window.Lang = {
  current: localStorage.getItem('wms_lang') || 'en',
  set(lang) {
    this.current = lang;
    localStorage.setItem('wms_lang', lang);
    this.applyDir();
    if (window.App) App.route();
  },
  applyDir() {
    document.documentElement.lang = this.current;
    document.documentElement.dir = RTL_LANGS.includes(this.current) ? 'rtl' : 'ltr';
  },
};
Lang.applyDir();

/**
 * Translate an English string into the active language.
 *
 * The English string IS the key, so an untranslated string falls back to itself
 * and the page still renders. That fallback is why the gap grew unnoticed: an
 * Arabic user saw English and nothing anywhere said so. Every miss is now
 * recorded in window.__i18nMisses, which is what turns silent degradation into
 * a number a test can assert on.
 *
 * `vars` fills {name} placeholders, so a sentence with a count in the middle can
 * be a single key. Without it those sentences cannot be translated at all —
 * they get assembled from fragments at the call site, and Arabic does not put
 * the fragments in the same order.
 */
function t(s, vars) {
  const dict = I18N[Lang.current];
  let out = dict && dict[s];
  if (out === undefined) {
    if (Lang.current !== 'en') {
      if (!window.__i18nMisses) window.__i18nMisses = new Set();
      window.__i18nMisses.add(s);
    }
    out = s;
  }
  return vars ? out.replace(/\{(\w+)\}/g, (m, k) => (k in vars ? vars[k] : m)) : out;
}
window.t = t;

/**
 * Theme preference (light / dark). Persists in localStorage; switching
 * re-renders the app so charts pick up the theme-aware palette.
 */
window.Theme = {
  current: localStorage.getItem('wms_theme') || 'light',
  apply() { document.documentElement.dataset.theme = this.current; },
  set(theme) {
    this.current = theme;
    localStorage.setItem('wms_theme', theme);
    this.apply();
    if (window.App) App.route();
  },
  toggle() { this.set(this.current === 'dark' ? 'light' : 'dark'); },
};
Theme.apply();

File: wms flutter application/lib/core/i18n.dart
/// Lightweight i18n mirroring the web app: English strings are the keys and
/// [t] falls back to the key itself, so untranslated screens keep working.
/// The active language lives in [I18n.current]; Session persists it and
/// notifies the app to rebuild (Arabic also flips the app to RTL).
library;

class I18n {
  static String current = 'en';

  static const Map<String, Map<String, String>> _dict = {
    'ar': {
      // menu / sections
      'Home': 'الرئيسية', 'Good morning': 'صباح الخير', 'Good afternoon': 'مساء الخير',
      'Good evening': 'مساء الخير', 'Pick a process to get started': 'اختر عملية للبدء',
      'Dashboard': 'لوحة التحكم', 'AI Stock Analytics': 'تحليلات المخزون الذكية',
      'Notifications': 'الإشعارات', 'Create Request': 'إنشاء طلب', 'Requests': 'الطلبات',
      'Approvals': 'الموافقات', 'ERP Operator': 'مشغل ERP',
      'Bin & Batch Assign': 'تخصيص الموقع والدفعة', 'Picker Assignment': 'تعيين المنفذ',
      'My Picking Tasks': 'مهام الالتقاط', 'Goods Issue Posting': 'ترحيل الصرف',
      'Goods Receipt': 'استلام البضائع', 'Quality Inspection': 'فحص الجودة',
      'Batch Tracking': 'تتبع الدفعات', 'Expiry Alerts': 'تنبيهات الصلاحية',
      'Cycle Counting': 'الجرد الدوري', 'Materials': 'المواد', 'Users': 'المستخدمون',
      'Audit Trail': 'سجل التدقيق',
      'General': 'عام', 'Material Requests': 'طلبات المواد',
      'Warehouse Execution': 'تنفيذ المستودع', 'Receiving & Quality': 'الاستلام والجودة',
      'Master Data & Admin': 'البيانات الرئيسية والإدارة',
      // common
      'Settings': 'الإعدادات', 'Server': 'الخادم', 'Server URL': 'رابط الخادم',
      'Save': 'حفظ', 'Account': 'الحساب', 'Sign out': 'تسجيل الخروج',
      'Login': 'تسجيل الدخول', 'Email': 'البريد الإلكتروني', 'Password': 'كلمة المرور',
      'Language': 'اللغة', 'Theme': 'المظهر', 'System': 'النظام', 'Light': 'فاتح', 'Dark': 'داكن',
      'Appearance': 'المظهر واللغة', 'Name': 'الاسم', 'Role': 'الدور', 'Permissions': 'الصلاحيات',
      'Submit': 'إرسال', 'Cancel': 'إلغاء', 'Confirm': 'تأكيد', 'Back': 'رجوع',
      'Approve': 'موافقة', 'Reject': 'رفض', 'Search': 'بحث', 'Status': 'الحالة',
      'Server URL saved.': 'تم حفظ رابط الخادم.',
      // R3: reallocation / inventory / shipping / scanning
      'Stock Reallocation': 'إعادة توزيع المخزون', 'Shipping & Outbound': 'الشحن والصادر',
      'Delivery & Dispatch': 'التسليم والإرسال', 'Physical Inventory': 'الجرد الفعلي',
      'Press back again to exit': 'اضغط رجوع مرة أخرى للخروج',
      'Scan QR / barcode': 'مسح QR / باركود', 'Scan with camera': 'مسح بالكاميرا',
      'Point the camera at the batch or bin label': 'وجّه الكاميرا نحو ملصق الدفعة أو الموقع',
      'Scan Bin': 'مسح الموقع', 'Scan bin QR / barcode': 'مسح رمز QR / باركود الموقع',
      'Scan bin label': 'مسح ملصق الموقع', 'Or enter bin code': 'أو أدخل رمز الموقع',
      'Go': 'انتقال', 'Could not look up this bin.': 'تعذّر العثور على هذا الموقع.',
      'Occupied': 'مشغول', 'Empty': 'فارغ', 'Contents': 'المحتويات',
      'Batch / lot detail': 'تفاصيل الدفعة', 'Batch': 'دفعة', 'Expiry': 'الصلاحية',
      'Quality': 'الجودة', 'Hazard': 'خطر', 'Temperature controlled': 'محكوم الحرارة',
      'Quality restricted': 'مقيد الجودة',
      'Reallocate': 'إعادة توزيع', 'Move stock': 'نقل المخزون',
      'New reallocation': 'إعادة توزيع جديدة', 'No reallocations yet': 'لا توجد عمليات إعادة توزيع',
      'Quantity to move': 'الكمية المنقولة', 'Target warehouse': 'المستودع الهدف',
      'Target bin': 'الموقع الهدف', 'Reason': 'السبب',
      'Count session': 'جلسة جرد', 'New count session': 'جلسة جرد جديدة',
      'Annual inventory': 'جرد سنوي', 'Periodic inventory': 'جرد دوري',
      'Counted quantity': 'الكمية المعدودة', 'Record count': 'تسجيل العد',
      'Post adjustments': 'ترحيل التسويات', 'Move to review': 'إلى المراجعة',
      'Cancel session': 'إلغاء الجلسة', 'Recount': 'إعادة العد', 'Count': 'عد',
      'Delivery order': 'أمر تسليم', 'New delivery order': 'أمر تسليم جديد',
      'Pack': 'تعبئة', 'Load': 'تحميل', 'Dispatch': 'إرسال', 'Deliver': 'تسليم',
      'Ship to': 'الوجهة', 'Carrier': 'الناقل', 'Received by': 'المستلم',
      'Confirm delivery (POD)': 'تأكيد التسليم', 'Confirm delivery': 'تأكيد التسليم',
      'Create shipment': 'إنشاء شحنة', 'No shipments yet': 'لا توجد شحنات',
      'Requester': 'مقدم الطلب', 'Department': 'القسم', 'Project': 'المشروع',
      'Cost Center': 'مركز التكلفة', 'Priority': 'الأولوية', 'Required date': 'التاريخ المطلوب',
      'Purpose': 'الغرض',
      // R4: reverse workflow / dashboard drill-through
      'Reverse one step': 'الرجوع خطوة للوراء',
      'Reason (releases any reservation, allocation or picking task this stage holds)':
          'السبب(ɏفرج عن أي حجز أو تخصيص أو مهمة التقاط تحتفظ بها هذه المرحلة)',
      'Request reversed.': 'تم إرجاع الطلب.',
    },
    'fr': {
      'Home': 'Accueil', 'Good morning': 'Bonjour', 'Good afternoon': 'Bonjour',
      'Good evening': 'Bonsoir', 'Pick a process to get started': 'Choisissez un processus pour commencer',
      'Dashboard': 'Tableau de bord', 'AI Stock Analytics': 'Analyse IA du stock',
      'Notifications': 'Notifications', 'Create Request': 'Créer une demande', 'Requests': 'Demandes',
      'Approvals': 'Approbations', 'ERP Operator': 'Opérateur ERP',
      'Bin & Batch Assign': 'Affectation casier/lot', 'Picker Assignment': 'Affectation préparateur',
      'My Picking Tasks': 'Mes préparations', 'Goods Issue Posting': 'Sortie de marchandises',
      'Goods Receipt': 'Réception', 'Quality Inspection': 'Inspection qualité',
      'Batch Tracking': 'Suivi des lots', 'Expiry Alerts': 'Alertes de péremption',
      'Cycle Counting': 'Comptage tournant', 'Materials': 'Articles', 'Users': 'Utilisateurs',
      'Audit Trail': "Piste d'audit",
      'General': 'Général', 'Material Requests': 'Demandes de matériel',
      'Warehouse Execution': 'Exécution entrepôt', 'Receiving & Quality': 'Réception et qualité',
      'Master Data & Admin': 'Données de base et admin',
      'Settings': 'Paramètres', 'Server': 'Serveur', 'Server URL': 'URL du serveur',
      'Save': 'Enregistrer', 'Account': 'Compte', 'Sign out': 'Déconnexion',
      'Login': 'Connexion', 'Email': 'E-mail', 'Password': 'Mot de passe',
      'Language': 'Langue', 'Theme': 'Thème', 'System': 'Système', 'Light': 'Clair', 'Dark': 'Sombre',
      'Appearance': 'Apparence', 'Name': 'Nom', 'Role': 'Rôle', 'Permissions': 'Autorisations',
      'Submit': 'Soumettre', 'Cancel': 'Annuler', 'Confirm': 'Confirmer', 'Back': 'Retour',
      'Approve': 'Approuver', 'Reject': 'Rejeter', 'Search': 'Rechercher', 'Status': 'Statut',
      'Server URL saved.': 'URL du serveur enregistrée.',
      // R3: reallocation / inventory / shipping / scanning
      'Stock Reallocation': 'Réaffectation du stock', 'Shipping & Outbound': 'Expédition',
      'Delivery & Dispatch': 'Livraison et envoi', 'Physical Inventory': 'Inventaire physique',
      'Press back again to exit': 'Appuyez à nouveau pour quitter',
      'Scan QR / barcode': 'Scanner QR / code-barres', 'Scan with camera': 'Scanner avec la caméra',
      'Point the camera at the batch or bin label': "Visez l'étiquette du lot ou du casier",
      'Scan Bin': 'Scanner un casier', 'Scan bin QR / barcode': 'Scanner le QR / code-barres du casier',
      'Scan bin label': "Scanner l'étiquette du casier", 'Or enter bin code': 'Ou saisir le code du casier',
      'Go': 'Aller', 'Could not look up this bin.': 'Impossible de trouver ce casier.',
      'Occupied': 'Occupé', 'Empty': 'Vide', 'Contents': 'Contenu',
      'Batch / lot detail': 'Détail du lot', 'Batch': 'Lot', 'Expiry': 'Expiration',
      'Quality': 'Qualité', 'Hazard': 'Danger', 'Temperature controlled': 'Température contrôlée',
      'Quality restricted': 'Restriction qualité',
      'Reallocate': 'Réaffecter', 'Move stock': 'Déplacer le stock',
      'New reallocation': 'Nouvelle réaffectation', 'No reallocations yet': 'Aucune réaffectation',
      'Quantity to move': 'Quantité à déplacer', 'Target warehouse': 'Entrepôt cible',
      'Target bin': 'Casier cible', 'Reason': 'Motif',
      'Count session': 'Session de comptage', 'New count session': 'Nouvelle session de comptage',
      'Annual inventory': 'Inventaire annuel', 'Periodic inventory': 'Inventaire périodique',
      'Counted quantity': 'Quantité comptée', 'Record count': 'Enregistrer le comptage',
      'Post adjustments': 'Valider les ajustements', 'Move to review': 'Passer en revue',
      'Cancel session': 'Annuler la session', 'Recount': 'Recompter', 'Count': 'Compter',
      'Delivery order': 'Ordre de livraison', 'New delivery order': 'Nouvel ordre de livraison',
      'Pack': 'Emballer', 'Load': 'Charger', 'Dispatch': 'Expédier', 'Deliver': 'Livrer',
      'Ship to': 'Destination', 'Carrier': 'Transporteur', 'Received by': 'Reçu par',
      'Confirm delivery (POD)': 'Confirmer la livraison', 'Confirm delivery': 'Confirmer la livraison',
      'Create shipment': "Créer l'expédition", 'No shipments yet': 'Aucune expédition',
      'Requester': 'Demandeur', 'Department': 'Département', 'Project': 'Projet',
      'Cost Center': 'Centre de coûts', 'Priority': 'Priorité', 'Required date': 'Date requise',
      'Purpose': 'Objet',
      // R4: reverse workflow / dashboard drill-through
      'Reverse one step': 'Revenir en arrière d\'une étape',
      'Reason (releases any reservation, allocation or picking task this stage holds)':
          'Motif (libère toute réservation, allocation ou tâche de préparation détenue à ce stade)',
      'Request reversed.': 'Demande renvoyée en arrière.',
    },
  };

  static bool get isRtl => current == 'ar';
  static String t(String s) => _dict[current]?[s] ?? s;
}

/// Translate [s] into the active language (falls back to the English key).
String t(String s) => I18n.t(s);
